Electron/src/RAM/Autonomy/components/Activity.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from "react"
+import CSS from "csstype"
+import { rovecomm } from "../../../Core/RoveProtocol/Rovecomm"
+
+const container: CSS.Properties = {
+  display: "flex",
+  flexDirection: "column",
+  fontFamily: "arial",
+  borderTopWidth: "28px",
+  borderColor: "#990000",
+  borderBottomWidth: "2px",
+  borderStyle: "solid",
+  whiteSpace: "pre-wrap",
+  overflow: "scroll",
+  height: "320px",
+}
+const label: CSS.Properties = {
+  marginTop: "-10px",
+  position: "relative",
+  top: "24px",
+  left: "3px",
+  fontFamily: "arial",
+  fontSize: "16px",
+  zIndex: 1,
+  color: "white",
+}
+
+interface IProps {
+  style?: CSS.Properties
+}
+
+interface IState {
+  ActivityText: string
+  backgroundColor: string
+}
+
+class Activity extends Component<IProps, IState> {
+  constructor(props: IProps) {
+    super(props)
+    this.state = {
+      ActivityText: "",
+      backgroundColor: "white",
+    }
+
+    this.ReachedMarker = this.ReachedMarker.bind(this)
+
+    rovecomm.on("AutonomyActivity", (data: any) => this.Log(data))
+    rovecomm.on("CurrentLog", (data: any) => this.Log(data))
+    rovecomm.on("ReachedMarker", this.ReachedMarker)
+  }
+
+  Log(data: string): void {
+    let text = this.state.ActivityText
+    text += `${new Date().toLocaleTimeString()}: ${data} \n` // set this to be that variable and time and newline
+    this.setState({
+      ActivityText: text,
+    })
+  }
+
+  ReachedMarker(): void {
+    this.setState({ backgroundColor: "green" })
+    this.Log("Reached waypoint!")
+    const reachInterval = setInterval(() => {
+      const backgroundColor: string = this.state.backgroundColor === "green" ? "white" : "green"
```

Comment:
can we/do we use the background colors in the same way on any other components? i feel like it could be cool

Reply:
The intent of this function is to strobe the background several times between green and white, and ensure it always ends on white. This is a requirement of SAR, as when the rover reaches an AR tag and stops it must be indicated both on rover and on basestation. In most cases throughout basestation a direct toggle of background color isn't useful (especially not useful enough to abstract this out into any kind of converter), but there is a [ColorConverter](https://github.com/MissouriMRDT/BaseStation_Software/blob/dev/Electron/src/Core/ColorConverter.tsx) in Core that takes in several parameters to allow responsive background colors on the screen.